psychopy/app/coder/coder.py

```diff
@@ -2907,18 +2908,22 @@ def onURL(self, evt):
         self.gotoLine(filename, lineNumber)
 
     def onUnitTests(self, evt=None):
-        """Show the unit tests frame
-        """
+        """Show the unit tests frame"""
         if self.unitTestFrame:
             self.unitTestFrame.Raise()
         else:
             self.unitTestFrame = UnitTestFrame(app=self.app)
         # UnitTestFrame.Show()
 
     def onPavloviaSync(self, evt=None):
-        print("Please sync your project from Builder")
-        # TODO: Allow user to sync project from coder
-        pass
+        """Push changes to project repo, or create new proj if proj is None"""
+        try:
+            self.project = pavlovia.getProject(self.currentDoc.filename)
```

Comment:
Does it need a try...except? What's the error? Ideally you should be able to call `getProject(filename)` and then just test if the result was None, rather than needing an error test.

Reply:
Ah yes, the exception was there to catch connection errors, but not explicitly. I could add ```requests.exceptions.ConnectionError```?